plugins/flutter_aepcore/ios/Classes/FlutterAEPCoreDataBridge.m

```diff
@@ -89,4 +91,48 @@ + (NSDictionary *)sanitizeDictionaryToContainClass: (Class) type WithDictionary:
     return sanitizedDict;
 }
 
++ (AEPInitOptions *)initOptionsFromMap:(id)map {
+    if (![map isKindOfClass:[FlutterMethodCall class]]) {
+        return nil;
+    }
+
+    FlutterMethodCall *call = (FlutterMethodCall *)map;
+    NSDictionary *arguments = call.arguments;
+    if (![arguments isKindOfClass:[NSDictionary class]]) {
+        return nil;
+    }
+
+    NSDictionary *initOptionsMap = arguments[@"initOptions"];
+    if (![initOptionsMap isKindOfClass:[NSDictionary class]]) {
+        return nil;
+    }
+
+    NSString *appId = initOptionsMap[@"appId"];
+    NSNumber *lifecycleAutomaticTrackingEnabled = initOptionsMap[@"lifecycleAutomaticTrackingEnabled"];
+    NSDictionary *lifecycleAdditionalContextData = initOptionsMap[@"lifecycleAdditionalContextData"];
+    NSString *appGroup = initOptionsMap[@"appGroup"];
+
+    AEPInitOptions *initOptions;
+    if (appId != nil && ![appId isKindOfClass:[NSNull class]]) {
```

Comment:
Instead of testing if `appId` is not NSNull class, can you check if it is NSString class?
```if (appId != nil && [appId isKindOfClass:[NSString class]])```